applications/crossbar/src/modules_v2/cb_devices_v2.erl

```diff
@@ -870,15 +863,17 @@ remove_if_mobile(MDN, Context) ->
     Normalized = knm_converters:normalize(MDN),
     case knm_number:get(Normalized) of
         {'ok', Number} ->
+            PN = knm_number:phone_number(Number),
+            IsMdnCarrier = ?CARRIER_MDN =:= knm_phone_number:module_name(PN),
             case kz_json:get_ne_value(<<"mobile">>, knm_number:to_public_json(Number)) of
-                'undefined' -> Context;
+                'undefined' when not IsMdnCarrier -> Context;
                 Mobile ->
```

Comment:
If mobile is not undefined, so device is mobile anyway. I'm checking module_name here only for cases that it's a mobile number but carrier sets to different carrier name because maybe it's an old number or something.